src/resmom/stage_func.c

```diff
@@ -482,10 +482,20 @@ is_direct_write(job *pjob, enum job_file which, char *path, int *direct_write_po
 
 	if (local_or_remote(&p) == 1) {
 		*direct_write_possible = 0;
-		sprintf(log_buffer,
+		if (pjob->ji_hosts != NULL) {
+			sprintf(log_buffer,
```

Comment:
I suggest doing snprintf() here to make sure we don't overflow.

Reply:
I'll use log_eventf().